### ISSUE TITLE
new Blob([buffer]) crashes renderer when buffer = Buffer.from('...', 'base64) and buffer.length >= 4096

### ISSUE THREAD

**Issue by dbakuntsev** [electron/electron#20750]
### Preflight Checklist

* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:**
  * 7.0.0
* **Operating System:**
  * Windows 8.1
* **Last Known Working Electron version:**
  * 3.0.3 (we're in process of upgrading from this version to the latest, so it is unclear as to when exactly the undesired behavior was introduced)

### Expected Behavior
Expecting no crash when the following code is executed:

```
const makeArray = (n, min = 1, max = Number.MAX_SAFE_INTEGER) => [...Array(n).keys()].map(i => (i + min) % max);
const encodeBase64 = (array) => btoa(String.fromCharCode(...array));

let buffer = Buffer.from(encodeBase64(makeArray(1024 * 4, 0, 256)), 'base64');
let blob = new Blob([buffer], { type: 'image/png' });
```

### Actual Behavior
The renderer process crashes while executing ```new Blob([buffer], { type: 'image/png' })``` statement. The Electron app window is blanked out, and if the Developer Tools is open, the following pop-up message is displayed: "DevTools was disconnected from the page". No Developer Tools Console messages. No messages in the OS console, either.

### To Reproduce
Clone the electron-quick-start repository, and insert the JavaScript code above into renderer.js file. Don't forget to set nodeIntegration = true for the main window, otherwise Buffer won't be found.

Electron Fiddle gist of the same: https://gist.github.com/70e141736d51fb0a6936c110b6f7c9f5

### Additional Information
This behavior is related to ```Buffer.from('...', 'base64')``` - that is, Base64 decoding - because providing the Blob constructor a buffer created by other means seems to work fine.

Also, the Base64-encoded string in the code above is just a simple way to reproduce the problem. The crash occurs when legitimate Base64-encoded images are processed as well.

**Comment by miniak** [comment]
It still works fine in Electron 4.2.12, it crashes in Electron 5.0.11

**Comment by miniak** [comment]
```
[29591:1027/103843.251648:FATAL:v8_array_buffer.cc(74)] Check failed: wrapper_type. 

0   libbase.dylib                       0x000000010e7f208f base::debug::CollectStackTrace(void**, unsigned long) + 31

1   libbase.dylib                       0x000000010e487b4b base::debug::StackTrace::StackTrace(unsigned long) + 75

2   libbase.dylib                       0x000000010e487b9d base::debug::StackTrace::StackTrace(unsigned long) + 29

3   libbase.dylib                       0x000000010e47bf28 base::debug::StackTrace::StackTrace() + 40

4   libbase.dylib                       0x000000010e4a6c13 logging::LogMessage::~LogMessage() + 131

5   libbase.dylib                       0x000000010e46d9f5 logging::LogMessage::~LogMessage() + 21

6   libblink_core.dylib                 0x0000000164662ed7 blink::V8ArrayBuffer::ToImpl(v8::Local<v8::Object>) + 663

7   libblink_core.dylib                 0x0000000164674a23 blink::V8Uint8Array::ToImpl(v8::Local<v8::Object>) + 739

8   libblink_core.dylib                 0x000000016466378c blink::V8ArrayBufferView::ToImpl(v8::Local<v8::Object>) + 588

9   libblink_core.dylib                 0x00000001646641f6 blink::V8ArrayBufferView::ToImplWithTypeCheck(v8::Isolate*, v8::Local<v8::Value>) + 230

10  libblink_core.dylib                 0x00000001641a11fe blink::NotShared<blink::DOMArrayBufferView> blink::ToNotShared<blink::NotShared<blink::DOMArrayBufferView> >(v8::Isolate*, v8::Local<v8::Value>, blink::ExceptionState&) + 62

11  libblink_core.dylib                 0x00000001641a4bff blink::V8ArrayBufferOrArrayBufferViewOrBlobOrUSVString::ToImpl(v8::Isolate*, v8::Local<v8::Value>, blink::ArrayBufferOrArrayBufferViewOrBlobOrUSVString&, blink::UnionTypeConversionMode, blink::ExceptionState&) + 815

12  libblink_core.dylib                 0x00000001641a5171 blink::NativeValueTraits<blink::ArrayBufferOrArrayBufferViewOrBlobOrUSVString, void>::NativeValue(v8::Isolate*, v8::Local<v8::Value>, blink::ExceptionState&) + 81

13  libblink_core.dylib                 0x00000001644737b0 blink::NativeValueTraits<blink::IDLSequence<blink::ArrayBufferOrArrayBufferViewOrBlobOrUSVString>, void>::ConvertSequenceFast(v8::Isolate*, v8::Local<v8::Array>, blink::ExceptionState&, blink::HeapVector<blink::ArrayBufferOrArrayBufferViewOrBlobOrUSVString, 0u>&) + 640

14  libblink_core.dylib                 0x0000000164473291 blink::NativeValueTraits<blink::IDLSequence<blink::ArrayBufferOrArrayBufferViewOrBlobOrUSVString>, void>::NativeValue(v8::Isolate*, v8::Local<v8::Value>, blink::ExceptionState&) + 481

15  libblink_core.dylib                 0x0000000164449aa7 blink::blob_v8_internal::Constructor(v8::FunctionCallbackInfo<v8::Value> const&) + 2055

16  libblink_core.dylib                 0x0000000164449271 blink::blob_v8_internal::ConstructorCallback(v8::FunctionCallbackInfo<v8::Value> const&) + 401

17  libv8.dylib                         0x000000018439008e v8::internal::FunctionCallbackArguments::Call(v8::internal::CallHandlerInfo) + 1198

18  libv8.dylib                         0x0000000184322719 v8::internal::MaybeHandle<v8::internal::Object> v8::internal::(anonymous namespace)::HandleApiCallHelper<true>(v8::internal::Isolate*, v8::internal::Handle<v8::internal::HeapObject>, v8::internal::Handle<v8::internal::HeapObject>, v8::internal::Handle<v8::internal::FunctionTemplateInfo>, v8::internal::Handle<v8::internal::Object>, v8::internal::BuiltinArguments) + 1273

19  libv8.dylib                         0x0000000184321554 v8::internal::Builtin_Impl_HandleApiCall(v8::internal::BuiltinArguments, v8::internal::Isolate*) + 660

20  libv8.dylib                         0x000000018432102f v8::internal::Builtin_HandleApiCall(int, unsigned long*, v8::internal::Isolate*) + 127

21  libv8.dylib                         0x0000000185306640 Builtins_CEntry_Return1_DontSaveFPRegs_ArgvOnStack_BuiltinExit + 64

Task trace:

0   libipc.dylib                        0x0000000114cdbf03 IPC::(anonymous namespace)::ChannelAssociatedGroupController::Accept(mojo::Message*) + 1283

IPC message handler context: 0x78C1A58A
```

**Comment by miniak** [comment]
https://cs.chromium.org/chromium/src/out/Debug/gen/third_party/blink/renderer/bindings/core/v8/v8_array_buffer.cc?sq=package:chromium&g=0&l=74

**Comment by yashrajpchavda** [comment]
Any update on this? I am in the middle of upgrading electron from 3.x to 7.x and facing this issue. By the way, it happens on mac machines as well.

**Comment by burningTyger** [comment]
The issue that references this one here (#18943) has been fixed with 8.0.0 Please test if it works for you too.

**Comment by yashrajpchavda** [comment]
I just checked on electron 8.0.0, but unfortunately, it does not seem to be fixed. The app still crashes when it executes the following function

```
b64toBlob = (b64Data, contentType = '') =>
			new Blob([Buffer.from(b64Data, 'base64')], { type: contentType });
```

**Comment by manavmehta** [comment]
@sofianguy 
For [Zulip Desktop](https://github.com/zulip/zulip-desktop/), it didn't get fixed in electron 8.4.0. 
And this is annoying some of our users a lot lately. Any suggestions?
Refer these conversations for more on the bug
[Thread 1](https://chat.zulip.org/#narrow/stream/16-desktop/topic/crash.20on.20edit.20message/near/917499)
[Thread 2](https://chat.zulip.org/#narrow/stream/9-issues/topic/Grey.20Window.20on.20macOS/near/934470)

**Comment by tzarebczan** [comment]
We're also seeing this issue - any updates or workarounds?

**Comment by seanyesmunt** [comment]
The https://github.com/lbryio/lbry-desktop app is seeing a crash when using the new File constructor

`new File([imageBuffer], 'thumbnail.png', { type: 'image/png' });`

I think it's related to this issue. It crashes in the same way. 

On Electron 9.3.1

**Comment by robertying** [comment]
I'm seeing a similar crash when using `Buffer`/`Uint8Array`/`TypedArray` with Electron in the renderer process. But it's not related to base64 or any sort.

Since this is the top issue when I googled with `electron crash blob`, I think I'll just leave a comment here in case anyone else googles into this issue and finds themselves in a similar position.

---

TL;DR: try to make a copy of your data because they may get GC'd unexpectedly.

---

My problem is that using a native addon ([kinect-azure](https://github.com/wouterverweirder/kinect-azure)) to read images works in the main process, but crashed in the renderer, only when I try to access the Uint8Array data. For example, `Buffer.from(imageData)` or `new Blob([imageData])`, where `imageData` is a Uint8Array.

Reading images directly in the main process is ok.

I've read some more issues and come up with the idea why it crashed in the renderer:

- https://github.com/electron/electron/issues/18943#issuecomment-708879736 is where I get the idea to copy the data.
- https://github.com/realm/realm-js/issues/1197 and https://github.com/electron/electron/issues/7732 tell me the possibility that there can be conflicts on GC between Node and the browser.

Even if you are holding a buffer data in the browser context, it can still be GC'd by Node. So make sure to make a copy if this happens.

**Comment by nornagon** [comment]
Here's a fiddle that reproduces the issue: https://gist.github.com/b99915ff3805c4f95a43073a1e4ec8da

**Comment by nornagon** [comment]
<details>
<summary>Full stack trace of crashed thread</summary>

```
Thread 0 (crashed)
 0  Electron Framework!blink::V8ArrayBuffer::ToImpl(v8::Local<v8::Object>) [v8_array_buffer.cc : 77 + 0x0]
    rax = 0x0000000000000000   rdx = 0x000000010b40e948
    rcx = 0x000000010b40e8f4   rbx = 0xffffffff00000000
    rsi = 0x0000000000000000   rdi = 0x0000000000000424
    rbp = 0x00007ffeea42c380   rsp = 0x00007ffeea42c300
     r8 = 0x000000000000000f    r9 = 0x0000000000000100
    r10 = 0x0000002784414100   r11 = 0x000000400000a7a8
    r12 = 0x00007ffeea42c460   r13 = 0x00007ffeea42c560
    r14 = 0x00007fd5ba047b80   r15 = 0x00007ffeea42c3f0
    rip = 0x000000011000993b
    Found by: given as instruction pointer in context
 1  Electron Framework!blink::V8Uint8Array::ToImpl(v8::Local<v8::Object>) [v8_uint8_array.cc : 73 + 0x8]
    rbp = 0x00007ffeea42c3b0   rsp = 0x00007ffeea42c390
    rip = 0x000000011000a868
    Found by: previous frame's frame pointer
 2  Electron Framework!blink::NotShared<blink::DOMArrayBufferView> blink::ToNotShared<blink::NotShared<blink::DOMArrayBufferView> >(v8::Isolate*, v8::Local<v8::Value>, blink::ExceptionState&) [v8_binding_for_core.h : 479 + 0xb]
    rbp = 0x00007ffeea42c3e0   rsp = 0x00007ffeea42c3c0
    rip = 0x000000010ffe833b
    Found by: previous frame's frame pointer
 3  Electron Framework!blink::V8ArrayBufferOrArrayBufferViewOrBlobOrUSVString::ToImpl(v8::Isolate*, v8::Local<v8::Value>, blink::ArrayBufferOrArrayBufferViewOrBlobOrUSVString&, blink::UnionTypeConversionMode, blink::ExceptionState&) [array_buffer_or_array_buffer_view_or_blob_or_usv_string.cc : 127 + 0xe]
    rbp = 0x00007ffeea42c420   rsp = 0x00007ffeea42c3f0
    rip = 0x000000010ffe8173
    Found by: previous frame's frame pointer
 4  Electron Framework!blink::NativeValueTraits<blink::IDLSequence<blink::IDLUnionNotINT<blink::ArrayBufferOrArrayBufferViewOrBlobOrUSVString> >, void>::ConvertSequenceFast(v8::Isolate*, v8::Local<v8::Array>, blink::ExceptionState&, blink::HeapVector<blink::ArrayBufferOrArrayBufferViewOrBlobOrUSVString, 0u>&) [native_value_traits_impl.h : 1277 + 0x1a]
    rbp = 0x00007ffeea42c4d0   rsp = 0x00007ffeea42c430
    rip = 0x000000011004874d
    Found by: previous frame's frame pointer
 5  Electron Framework!blink::NativeValueTraits<blink::IDLSequence<blink::IDLUnionNotINT<blink::ArrayBufferOrArrayBufferViewOrBlobOrUSVString> >, void>::NativeValue(v8::Isolate*, v8::Local<v8::Value>, blink::ExceptionState&) [native_value_traits_impl.h : 788 + 0xe]
    rbp = 0x00007ffeea42c590   rsp = 0x00007ffeea42c4e0
    rip = 0x00000001100485a2
    Found by: previous frame's frame pointer
 6  Electron Framework!blink::(anonymous namespace)::ConstructorCallback(v8::FunctionCallbackInfo<v8::Value> const&) [native_value_traits.h : 119 + 0xb]
    rbp = 0x00007ffeea42c610   rsp = 0x00007ffeea42c5a0
    rip = 0x000000010c594d6c
    Found by: previous frame's frame pointer
 7  Electron Framework!v8::internal::FunctionCallbackArguments::Call(v8::internal::CallHandlerInfo) [api-arguments-inl.h : 158 + 0x5]
    rbp = 0x00007ffeea42c700   rsp = 0x00007ffeea42c620
    rip = 0x000000010b32a9b9
    Found by: previous frame's frame pointer
 8  Electron Framework!v8::internal::MaybeHandle<v8::internal::Object> v8::internal::(anonymous namespace)::HandleApiCallHelper<true>(v8::internal::Isolate*, v8::internal::Handle<v8::internal::HeapObject>, v8::internal::Handle<v8::internal::HeapObject>, v8::internal::Handle<v8::internal::FunctionTemplateInfo>, v8::internal::Handle<v8::internal::Object>, v8::internal::BuiltinArguments) [builtins-api.cc : 111 + 0xb]
    rbp = 0x00007ffeea42c7b0   rsp = 0x00007ffeea42c710
    rip = 0x000000010b32a2f1
    Found by: previous frame's frame pointer
 9  Electron Framework!v8::internal::Builtin_Impl_HandleApiCall(v8::internal::BuiltinArguments, v8::internal::Isolate*) [builtins-api.cc : 137 + 0x18]
    rbp = 0x00007ffeea42c840   rsp = 0x00007ffeea42c7c0
    rip = 0x000000010b329d7a
    Found by: previous frame's frame pointer
10  Electron Framework!Builtins_CEntry_Return1_DontSaveFPRegs_ArgvOnStack_BuiltinExit + 0x38
    rbp = 0x00007ffeea42c860   rsp = 0x00007ffeea42c850
    rip = 0x000000010b581058
    Found by: previous frame's frame pointer
11  Electron Framework!Builtins_JSBuiltinsConstructStub + 0x65
    rbp = 0x00007ffeea42c8c0   rsp = 0x00007ffeea42c870
    rip = 0x000000010b5106c5
    Found by: previous frame's frame pointer
12  Electron Framework!Builtins_ConstructHandler + 0x2bf
    rbp = 0x00007ffeea42c930   rsp = 0x00007ffeea42c8d0
    rip = 0x000000010b61101f
    Found by: previous frame's frame pointer
13  Electron Framework!Builtins_InterpreterEntryTrampoline + 0xd8
    rbp = 0x00007ffeea42c9c0   rsp = 0x00007ffeea42c940
    rip = 0x000000010b513698
    Found by: previous frame's frame pointer
14  Electron Framework!Builtins_InterpreterEntryTrampoline + 0xd8
    rbp = 0x00007ffeea42ca70   rsp = 0x00007ffeea42c9d0
    rip = 0x000000010b513698
    Found by: previous frame's frame pointer
15  Electron Framework!Builtins_InterpreterEntryTrampoline + 0xd8
    rbp = 0x00007ffeea42cb98   rsp = 0x00007ffeea42ca80
    rip = 0x000000010b513698
    Found by: previous frame's frame pointer
16  Electron Framework!Builtins_InterpreterEntryTrampoline + 0xd8
    rbp = 0x00007ffeea42cc28   rsp = 0x00007ffeea42cba8
    rip = 0x000000010b513698
    Found by: previous frame's frame pointer
17  Electron Framework!Builtins_InterpreterEntryTrampoline + 0xd8
    rbp = 0x00007ffeea42ccb8   rsp = 0x00007ffeea42cc38
    rip = 0x000000010b513698
    Found by: previous frame's frame pointer
18  Electron Framework!Builtins_InterpreterEntryTrampoline + 0xd8
    rbp = 0x00007ffeea42cd90   rsp = 0x00007ffeea42ccc8
    rip = 0x000000010b513698
    Found by: previous frame's frame pointer
19  Electron Framework!Builtins_InterpreterEntryTrampoline + 0xd8
    rbp = 0x00007ffeea42ce30   rsp = 0x00007ffeea42cda0
    rip = 0x000000010b513698
    Found by: previous frame's frame pointer
20  Electron Framework!Builtins_ArgumentsAdaptorTrampoline + 0xb8
    rbp = 0x00007ffeea42ce70   rsp = 0x00007ffeea42ce40
    rip = 0x000000010b50baf8
    Found by: previous frame's frame pointer
21  Electron Framework!Builtins_InterpreterEntryTrampoline + 0xd8
    rbp = 0x00007ffeea42cf08   rsp = 0x00007ffeea42ce80
    rip = 0x000000010b513698
    Found by: previous frame's frame pointer
22  Electron Framework!Builtins_InterpreterEntryTrampoline + 0xd8
    rbp = 0x00007ffeea42cf60   rsp = 0x00007ffeea42cf18
    rip = 0x000000010b513698
    Found by: previous frame's frame pointer
23  Electron Framework!Builtins_InterpreterEntryTrampoline + 0xd8
    rbp = 0x00007ffeea42cfc0   rsp = 0x00007ffeea42cf70
    rip = 0x000000010b513698
    Found by: previous frame's frame pointer
24  Electron Framework!Builtins_JSEntryTrampoline + 0x5b
    rbp = 0x00007ffeea42cfe8   rsp = 0x00007ffeea42cfd0
    rip = 0x000000010b5111bb
    Found by: previous frame's frame pointer
25  Electron Framework!Builtins_JSEntry + 0x78
    rbp = 0x00007ffeea42d050   rsp = 0x00007ffeea42cff8
    rip = 0x000000010b510f98
    Found by: previous frame's frame pointer
26  Electron Framework!v8::internal::(anonymous namespace)::Invoke(v8::internal::Isolate*, v8::internal::(anonymous namespace)::InvokeParams const&) [simulator.h : 142 + 0xb]
    rbp = 0x00007ffeea42d1b0   rsp = 0x00007ffeea42d060
    rip = 0x000000010b34aa99
    Found by: previous frame's frame pointer
27  Electron Framework!v8::internal::Execution::Call(v8::internal::Isolate*, v8::internal::Handle<v8::internal::Object>, v8::internal::Handle<v8::internal::Object>, int, v8::internal::Handle<v8::internal::Object>*) [execution.cc : 462 + 0x8]
    rbp = 0x00007ffeea42d240   rsp = 0x00007ffeea42d1c0
    rip = 0x000000010b34a2f9
    Found by: previous frame's frame pointer
28  Electron Framework!v8::Script::Run(v8::Local<v8::Context>) [api.cc : 2163 + 0x11]
    rbp = 0x00007ffeea42d360   rsp = 0x00007ffeea42d250
    rip = 0x000000010b30f54f
    Found by: previous frame's frame pointer
29  Electron Framework!blink::V8ScriptRunner::RunCompiledScript(v8::Isolate*, v8::Local<v8::Script>, blink::ExecutionContext*) [v8_script_runner.cc : 362 + 0x8]
    rbp = 0x00007ffeea42d480   rsp = 0x00007ffeea42d370
    rip = 0x000000010c4c0a5a
    Found by: previous frame's frame pointer
30  Electron Framework!blink::V8ScriptRunner::CompileAndRunScript(v8::Isolate*, blink::ScriptState*, blink::ExecutionContext*, blink::ScriptSourceCode const&, blink::KURL const&, blink::SanitizeScriptErrors, blink::ScriptFetchOptions const&, blink::mojom::V8CacheOptions, blink::V8ScriptRunner::RethrowErrorsOption) [v8_script_runner.cc : 440 + 0xf]
    rbp = 0x00007ffeea42d630   rsp = 0x00007ffeea42d490
    rip = 0x000000010c4c0f47
    Found by: previous frame's frame pointer
31  Electron Framework!blink::ScriptController::ExecuteScriptAndReturnValue(v8::Local<v8::Context>, blink::ScriptSourceCode const&, blink::KURL const&, blink::SanitizeScriptErrors, blink::ScriptFetchOptions const&) [script_controller.cc : 100 + 0x17]
    rbp = 0x00007ffeea42d6c0   rsp = 0x00007ffeea42d640
    rip = 0x000000010c4b75db
    Found by: previous frame's frame pointer
32  Electron Framework!blink::ScriptController::EvaluateScriptInMainWorld(blink::ScriptSourceCode const&, blink::KURL const&, blink::SanitizeScriptErrors, blink::ScriptFetchOptions const&, blink::ScriptController::ExecuteScriptPolicy) [script_controller.cc : 299 + 0x18]
    rbp = 0x00007ffeea42d730   rsp = 0x00007ffeea42d6d0
    rip = 0x000000010c4b791a
    Found by: previous frame's frame pointer
33  Electron Framework!blink::ClassicScript::RunScript(blink::LocalFrame*) [classic_script.cc : 43 + 0xb]
    rbp = 0x00007ffeea42d770   rsp = 0x00007ffeea42d740
    rip = 0x000000010c9c0446
    Found by: previous frame's frame pointer
34  Electron Framework!blink::PendingScript::ExecuteScriptBlockInternal(blink::Script*, blink::ScriptElementBase*, bool, bool, bool, base::TimeTicks, bool) [pending_script.cc : 264 + 0xc]
    rbp = 0x00007ffeea42d7d0   rsp = 0x00007ffeea42d780
    rip = 0x000000010c9c344b
    Found by: previous frame's frame pointer
35  Electron Framework!blink::PendingScript::ExecuteScriptBlock(blink::KURL const&) [pending_script.cc : 170 + 0x23]
    rbp = 0x00007ffeea42d860   rsp = 0x00007ffeea42d7e0
    rip = 0x000000010c9c3225
    Found by: previous frame's frame pointer
36  Electron Framework!blink::ScriptLoader::PrepareScript(WTF::TextPosition const&, blink::ScriptLoader::LegacyTypeSupport) [script_loader.cc : 914 + 0x10]
    rbp = 0x00007ffeea42da60   rsp = 0x00007ffeea42d870
    rip = 0x000000010c9c4ccb
    Found by: previous frame's frame pointer
37  Electron Framework!blink::HTMLParserScriptRunner::ProcessScriptElementInternal(blink::Element*, WTF::TextPosition const&) [html_parser_script_runner.cc : 609 + 0xd]
    rbp = 0x00007ffeea42db90   rsp = 0x00007ffeea42da70
    rip = 0x000000010c9c1d31
    Found by: previous frame's frame pointer
38  Electron Framework!blink::HTMLParserScriptRunner::ProcessScriptElement(blink::Element*, WTF::TextPosition const&) [html_parser_script_runner.cc : 332 + 0xe]
    rbp = 0x00007ffeea42dbc0   rsp = 0x00007ffeea42dba0
    rip = 0x000000010c9c1b7f
    Found by: previous frame's frame pointer
39  Electron Framework!blink::HTMLDocumentParser::RunScriptsForPausedTreeBuilder() [html_document_parser.cc : 539 + 0x8]
    rbp = 0x00007ffeea42dc40   rsp = 0x00007ffeea42dbd0
    rip = 0x000000010c8223e7
    Found by: previous frame's frame pointer
40  Electron Framework!blink::HTMLDocumentParser::ProcessTokenizedChunkFromBackgroundParser(std::__1::unique_ptr<blink::HTMLDocumentParser::TokenizedChunk, std::__1::default_delete<blink::HTMLDocumentParser::TokenizedChunk> >, bool*) [html_document_parser.cc : 780 + 0x8]
    rbp = 0x00007ffeea42dcf0   rsp = 0x00007ffeea42dc50
    rip = 0x000000010c822f14
    Found by: previous frame's frame pointer
41  Electron Framework!blink::HTMLDocumentParser::PumpPendingSpeculations() [html_document_parser.cc : 840 + 0x13]
    rbp = 0x00007ffeea42dd90   rsp = 0x00007ffeea42dd00
    rip = 0x000000010c8221ea
    Found by: previous frame's frame pointer
42  Electron Framework!blink::HTMLDocumentParser::ResumeParsingAfterPause() [html_document_parser.cc : 0 + 0x8]
    rbp = 0x00007ffeea42de10   rsp = 0x00007ffeea42dda0
    rip = 0x000000010c8243f8
    Found by: previous frame's frame pointer
43  Electron Framework!blink::HTMLDocumentParser::ExecuteScriptsWaitingForResources() [html_document_parser.cc : 1467 + 0x8]
    rbp = 0x00007ffeea42de60   rsp = 0x00007ffeea42de20
    rip = 0x00000001105bb783
    Found by: previous frame's frame pointer
44  Electron Framework!blink::TaskHandle::Runner::Run(blink::TaskHandle const&) [callback.h : 100 + 0x6]
    rbp = 0x00007ffeea42dea0   rsp = 0x00007ffeea42de70
    rip = 0x000000010b65aa57
    Found by: previous frame's frame pointer
45  Electron Framework!base::TaskAnnotator::RunTask(char const*, base::PendingTask*) [callback.h : 100 + 0x3]
    rbp = 0x00007ffeea42df80   rsp = 0x00007ffeea42deb0
    rip = 0x000000010ba5b049
    Found by: previous frame's frame pointer
46  Electron Framework!base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::DoWorkImpl(base::sequence_manager::LazyNow*) [thread_controller_with_message_pump_impl.cc : 332 + 0xf]
    rbp = 0x00007ffeea42e090   rsp = 0x00007ffeea42df90
    rip = 0x000000010ba68152
    Found by: previous frame's frame pointer
47  Electron Framework!base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::DoWork() [thread_controller_with_message_pump_impl.cc : 252 + 0xb]
    rbp = 0x00007ffeea42e0e0   rsp = 0x00007ffeea42e0a0
    rip = 0x000000010ba67d4e
    Found by: previous frame's frame pointer
48  Electron Framework!base::MessagePumpCFRunLoopBase::RunWork() [message_pump_mac.mm : 358 + 0x6]
    rbp = 0x00007ffeea42e120   rsp = 0x00007ffeea42e0f0
    rip = 0x000000010ba993e0
    Found by: previous frame's frame pointer
49  Electron Framework!base::mac::CallWithEHFrame(void () block_pointer) + 0xa
    rbp = 0x00007ffeea42e130   rsp = 0x00007ffeea42e130
    rip = 0x000000010ba95f7a
    Found by: previous frame's frame pointer
50  Electron Framework!base::MessagePumpCFRunLoopBase::RunWorkSource(void*) [message_pump_mac.mm : 334 + 0x5]
    rbp = 0x00007ffeea42e170   rsp = 0x00007ffeea42e140
    rip = 0x000000010ba98dbf
    Found by: previous frame's frame pointer
51  CoreFoundation!__CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 0x11
    rbp = 0x00007ffeea42e180   rsp = 0x00007ffeea42e180
    rip = 0x00007fff30baed52
    Found by: previous frame's frame pointer
52  CoreFoundation!__CFRunLoopDoSource0 + 0x67
    rbp = 0x00007ffeea42e1b0   rsp = 0x00007ffeea42e190
    rip = 0x00007fff30baecf1
    Found by: previous frame's frame pointer
53  CoreFoundation!__CFRunLoopDoSources0 + 0xd1
    rbp = 0x00007ffeea42e220   rsp = 0x00007ffeea42e1c0
    rip = 0x00007fff30baeb0b
    Found by: previous frame's frame pointer
54  CoreFoundation!__CFRunLoopRun + 0x39f
    rbp = 0x00007ffeea42ef30   rsp = 0x00007ffeea42e230
    rip = 0x00007fff30bad83a
    Found by: previous frame's frame pointer
55  CoreFoundation!CFRunLoopRunSpecific + 0x1ce
    rbp = 0x00007ffeea42efc0   rsp = 0x00007ffeea42ef40
    rip = 0x00007fff30bace3e
    Found by: previous frame's frame pointer
56  Foundation!-[NSRunLoop(NSRunLoop) runMode:beforeDate:] + 0xd4
    rbp = 0x00007ffeea42f000   rsp = 0x00007ffeea42efd0
    rip = 0x00007fff332481c8
    Found by: previous frame's frame pointer
57  Electron Framework!base::MessagePumpNSRunLoop::DoRun(base::MessagePump::Delegate*) [message_pump_mac.mm : 604 + 0xf]
    rbp = 0x00007ffeea42f050   rsp = 0x00007ffeea42f010
    rip = 0x000000010ba998e9
    Found by: previous frame's frame pointer
58  Electron Framework!base::MessagePumpCFRunLoopBase::Run(base::MessagePump::Delegate*) [message_pump_mac.mm : 149 + 0xc]
    rbp = 0x00007ffeea42f090   rsp = 0x00007ffeea42f060
    rip = 0x000000010ba98952
    Found by: previous frame's frame pointer
59  Electron Framework!base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::Run(bool, base::TimeDelta) [thread_controller_with_message_pump_impl.cc : 446 + 0x6]
    rbp = 0x00007ffeea42f0c0   rsp = 0x00007ffeea42f0a0
    rip = 0x000000010ba686d1
    Found by: previous frame's frame pointer
60  Electron Framework!base::RunLoop::Run() [run_loop.cc : 124 + 0x13]
    rbp = 0x00007ffeea42f170   rsp = 0x00007ffeea42f0d0
    rip = 0x000000010ba47c69
    Found by: previous frame's frame pointer
61  Electron Framework!content::RendererMain(content::MainFunctionParams const&) [renderer_main.cc : 256 + 0x5]
    rbp = 0x00007ffeea42f230   rsp = 0x00007ffeea42f180
    rip = 0x000000010ccd9197
    Found by: previous frame's frame pointer
62  Electron Framework!content::ContentMainRunnerImpl::Run(bool) [content_main_runner_impl.cc : 860 + 0x5]
    rbp = 0x00007ffeea42f2b0   rsp = 0x00007ffeea42f240
    rip = 0x000000010b6b476d
    Found by: previous frame's frame pointer
63  Electron Framework!content::RunContentProcess(content::ContentMainParams const&, content::ContentMainRunner*) [content_main.cc : 373 + 0x6]
    rbp = 0x00007ffeea42f4c0   rsp = 0x00007ffeea42f2c0
    rip = 0x000000010b2f2c40
    Found by: previous frame's frame pointer
64  Electron Framework!content::ContentMain(content::ContentMainParams const&) [content_main.cc : 399 + 0x8]
    rbp = 0x00007ffeea42f4f0   rsp = 0x00007ffeea42f4d0
    rip = 0x000000010b2f3112
    Found by: previous frame's frame pointer
65  Electron Framework!ElectronMain [electron_library_main.mm : 23 + 0x5]
    rbp = 0x00007ffeea42f590   rsp = 0x00007ffeea42f500
    rip = 0x000000010ae28348
    Found by: previous frame's frame pointer
66  Electron Helper (Renderer)!main [electron_main.cc : 274 + 0xb]
    rbp = 0x00007ffeea42f660   rsp = 0x00007ffeea42f5a0
    rip = 0x00000001057d3d20
    Found by: previous frame's frame pointer
67  libdyld.dylib!start + 0x1
    rbp = 0x00007ffeea42f678   rsp = 0x00007ffeea42f670
    rip = 0x00007fff6ac1ecc9
    Found by: previous frame's frame pointer
68  libdyld.dylib!start + 0x1
    rbp = 0x00007ffeea42f678   rsp = 0x00007ffeea42f678
    rip = 0x00007fff6ac1ecc9
    Found by: stack scanning
```

</details>

**Comment by nornagon** [comment]
Appears fixed in v12.0.0-nightly.20201112. Seems the fix happened in https://github.com/electron/electron/compare/v12.0.0-nightly.20200914...v12.0.0-nightly.20201013.

**Comment by nornagon** [comment]
You can work around this issue by passing a `Uint8Array` instead of a `Buffer` to the `Blob` constructor.
